tensorrt_llm/_torch/auto_deploy/transform/library/attention.py

```diff
@@ -414,91 +325,242 @@ def _grouped_attn_replacement_4(q, k, v, attn_mask, dropout_p, scale):
+            k1 = torch.randn(8, 1, 16, 64, device="cuda", dtype=torch.float16)
+            v1 = torch.randn(8, 1, 16, 64, device="cuda", dtype=torch.float16)
+            attn_mask = torch.randn(8, 1, 1, 16, device="cuda", dtype=torch.float16)
+            dropout = 0.12345
+            scale = 0.56789
+            n_rep = 7
+
+            dummy_args_1 = [q, k1, v1, n_rep, attn_mask, dropout, scale]
+            dummy_args_2 = [q, k1, v1, attn_mask, dropout, scale]
+
+            register_ad_pattern(
+                search_fn=_grouped_attn_pattern_1,
+                replace_fn=_grouped_attn_replacement_1,
+                patterns=patterns,
+                dummy_args=dummy_args_1,
+                scalar_workaround={"scale": scale, "dropout_p": dropout, "n_rep": n_rep},
+            )
+            register_ad_pattern(
+                search_fn=_grouped_attn_pattern_2,
+                replace_fn=_grouped_attn_replacement_2,
+                patterns=patterns,
+                dummy_args=dummy_args_2,
+                scalar_workaround={
+                    "scale": scale,
+                    "dropout_p": dropout,
+                },
+            )
+            register_ad_pattern(
+                search_fn=_grouped_attn_pattern_3,
+                replace_fn=_grouped_attn_replacement_3,
+                patterns=patterns,
+                dummy_args=dummy_args_1,
+                scalar_workaround={"scale": scale, "dropout_p": dropout, "n_rep": n_rep},
+            )
+            register_ad_pattern(
+                search_fn=_grouped_attn_pattern_4,
+                replace_fn=_grouped_attn_replacement_4,
+                patterns=patterns,
+                dummy_args=dummy_args_2,
+                scalar_workaround={
+                    "scale": scale,
+                    "dropout_p": dropout,
+                },
             )
 
-        # Check if need to update the output node to match the layout
-        if attention_layout == "bsnd":
-            # Add transpose for the output (from bsnd back to bnsd)
-            with graph.inserting_after(source_sdpa_node):
-                output_updated = graph.call_function(
-                    torch.ops.aten.transpose.int, args=(source_sdpa_node, 1, 2)
-                )
+        num_grouped_patterns = _apply_pattern(gm, "Grouped Attention", register_grouped_attention)
 
-            # Preserve fake tensor in meta["val"] for the transposed inputs
-            source_sdpa_node.meta["val"] = sdpa_node.meta["val"].transpose(1, 2).contiguous()
-            output_updated.meta["val"] = source_sdpa_node.meta["val"].transpose(1, 2)
-        elif attention_layout == "bnsd":
-            output_updated = source_sdpa_node
-        else:
-            raise ValueError(f"Unsupported attention layout: {attention_layout}")
+        info = TransformInfo(
+            skipped=False,
+            num_matches=num_grouped_patterns,
+            is_clean=True,
```

Comment:
```suggestion
            is_clean=False,
```

Same. Don't think we can assume is_clean after this transform